app/instance-initializers/ember-i18n.js

```diff
@@ -13,7 +14,14 @@ export default {
     }
     instance.container.lookup('service:i18n').set('locale', defaultLocale);
 
-    Ember.HTMLBars._registerHelper('t', helper);
+    if (legacyHelper != null) {
+      Ember.HTMLBars._registerHelper('t', legacyHelper);
+    }
+
+    if (Helper != null) {
+      instance.registry.register('helper:t', Helper);
```

Comment:
Correct. Robert is going to land https://github.com/emberjs/rfcs/pull/58 into 1.13 which will give you `{{t` without any explicit registration step.